service.php

```diff
@@ -216,8 +217,89 @@ function basic_ui() {
+            }
+
+            // Output basic form for collecting user/pass
+            /* translators: service name */
+            echo '<p>' . sprintf(__('Enter your username and password for accessing <strong>%s</strong>:', 'keyring'), $this->get_label()) . '</p>';
+            echo '<form method="post" action="">';
+            echo '<input type="hidden" name="service" value="' . esc_attr($this->get_name()) . '" />';
+            echo '<input type="hidden" name="action" value="manage" />';
+            wp_nonce_field('keyring-manage', 'kr_nonce', false);
+            wp_nonce_field('keyring-manage-' . $this->get_name(), 'nonce', false);
+            echo '<table class="form-table">';
+            echo '<tr><th scope="row">' . __('Username', 'keyring') . '</th>';
+            echo '<td><input type="text" name="username" value="" id="username" class="regular-text"></td></tr>';
+            echo '<tr><th scope="row">' . __('Password', 'keyring') . '</th>';
+            echo '<td><input type="password" name="password" value="" id="password" class="regular-text"></td></tr>';
+            echo '</table>';
+            echo '<p class="submitbox">';
+            echo '<input type="submit" name="submit" value="' . __('Verify Details', 'keyring') . '" id="submit" class="button-primary">';
+            echo '<a href="' . esc_url(Keyring_Util::admin_url(null, array('action' => 'services'))) . '" class="submitdelete" style="margin-left:2em;">' . __('Cancel', 'keyring') . '</a>';
+            echo '</p>';
+            echo '</form>';
+            echo '</div>';
+            ?>
+        <script type="text/javascript" charset="utf-8">
+            jQuery(document).ready(function () {
+                jQuery('#username').focus();
+            });
+        </script>
+        <?php
+
+        if (isset($_POST['username']) && isset($_POST['password'])) {
+            // HTTP Basic does not use Keyring_Request_Tokens, since there's only one step
+
+            $token = new Keyring_Access_Token(
+                $this->get_name(),
+                base64_encode(trim($_POST['username']) . ':' . trim($_POST['password']))
+            );
+            $this->set_token($token);
+
+            $meta = array_merge(
+                array(
+                    'username' => trim($_POST['username']),
+                    'basic' => true,
+                    apply_filters('keyring_access_token_meta', array(), $this->get_name(), $token, null, $this),
+                )
+            );
+
+            $access_token = new Keyring_Access_Token(
+                $this->get_name(),
+                $token,
+                $meta
+            );
+            $access_token = apply_filters('keyring_access_token', $access_token, array());
+
+            // If we didn't get a 401, then we'll assume it's OK
+            $this->store_token($access_token);
+            echo '<div class="updated"><p>' . __('Credentials saved.', 'keyring') . '</p></div>';
+            // todo redirect
```

Comment:
after saving the username and password, would be great to redirect to the token list page